test/inspector/inspector-helper.js

```diff
@@ -146,14 +155,22 @@ function TestSession(socket, harness) {
       if (consumed)
         buffer = buffer.slice(consumed);
     } while (consumed);
-  }).on('close', () => assert(this.expectClose_, 'Socket closed prematurely'));
+  }).on('close', () => {
+    assert(this.expectClose_, 'Socket closed prematurely');
+    this.closeCallback_ && this.closeCallback_();
```

Comment:
This test "framework" is really messy - I am trying to clean it up, but have not yet achieved the stage to create a pull request...

Currently the callback is set in sendCommandsAndExpectClose, e.g. when there's a need to handle connection closing and not just assert.